pkg/client/client.go

```diff
@@ -222,6 +224,36 @@ func (c *Client) GetProxy(name string) (*configv1.Proxy, error) {
 	return c.oscclient.ConfigV1().Proxies().Get(name, metav1.GetOptions{})
 }
 
+func (c *Client) GetAPIServerAuthenticationConfigmap() (*v1.ConfigMap, error) {
+	return c.kclient.CoreV1().ConfigMaps("kube-system").Get("extension-apiserver-authentication", metav1.GetOptions{})
+}
+
+func (c *Client) DeletePrometheusAdapterAPIAuthenticationSecret(newName string) error {
```

Comment:
Consider naming this `DeleteOld...` to be clear that is is not deleting the secret with the argument as the name, but an old secret that explicitly is NOT the argument. Otherwise, it sort of looks like it is designed to delete the specified resource.

Reply:
sounds good, I'll fix the naming

Reply:
The more I think about it, the more it seems to me that perhaps this func should be part of the prometheus-adapter task rather than the client package. It seems like very specific application logic encoding a core operator task than a general purpose client func. Thoughts?

Reply:
I am fine with factoring those methods into the task itself, agreed.